src/amqpvalue.c

```diff
@@ -3183,22 +3960,246 @@ static int encode_array(AMQPVALUE_ENCODER_OUTPUT encoder_output, void* context,
 
         if (result == 0)
         {
+            AMQP_TYPE array_type = AMQP_TYPE_UNKNOWN;
+
             for (i = 0; i < count; i++)
             {
-                if (amqpvalue_encode(items[i], encoder_output, context) != 0)
+                AMQP_VALUE_DATA* value_data = (AMQP_VALUE_DATA*)items[i];
+                if ((array_type != AMQP_TYPE_UNKNOWN) && (value_data->type != array_type))
```

Comment:
>if ((array_type != AMQP_TYPE_UNKNOWN) && (value_data->type != array_type)) [](start = 16, length = 74)

This should be caught at the time of creating the array.

Reply:
Thanks, I've refactored this out.